luafilesystem: update to 1.8.0

flyn-org opened the pull request:
Signed-off-by: W. Michael Petullo <mike@flyn.org>

Maintainer: me
Compile tested: mips_24kc master

Description:

neheb commented:
Uhh what? How is the source version unchanged?

neheb commented on lang/luafilesystem/Makefile:
```diff
@@ -8,13 +8,13 @@
+PKG_VERSION:=1.8.0
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
```
Not needed. .xz is also smaller and default.

karlp replied:
maybe so, but completely unrelated to this PR

neheb commented on lang/luafilesystem/Makefile:
```diff
@@ -8,13 +8,13 @@
-PKG_SOURCE_VERSION:=de87218e9798c4dd1a40d65403d99e9e82e1cfa0
+PKG_SOURCE_VERSION:=7c6e1b013caec0602ca4796df3b1d7253a2dd258
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
```
This is already default.

karlp replied:
yes, but it's completely out of scope of this PR, and it would be wrong to make unrelated changes under the banner of a version bump

neheb replied:
So when do you propose that this be fixed?

karlp replied:
at anyone's leisure, but not as part of this commit.  It does zero harm in anyway, it's simply redundant.

neheb commented on lang/luafilesystem/Makefile:
```diff
@@ -8,13 +8,13 @@
+PKG_MIRROR_HASH:=16d17c788b8093f2047325343f5e9b74cccb1ea96001e45914a58bbae8932495
 PKG_SOURCE_URL:=https://github.com/keplerproject/luafilesystem.git
-PKG_SOURCE_VERSION:=de87218e9798c4dd1a40d65403d99e9e82e1cfa0
+PKG_SOURCE_VERSION:=7c6e1b013caec0602ca4796df3b1d7253a2dd258
```
So the upstream tag is v1_8_0. There should be a way to turn PKG_VERSION to that with some shell magic.

flyn-org commented:
@neheb, I just force pushed a second commit that addresses your comments. I am grateful for your continued efforts to improve the quality and consistency of the package Makefiles.

neheb commented:
Mirror hash should be updated.

neheb commented:
WARNING: PKG_MIRROR_HASH does not match luafilesystem-1.8.0.tar.xz hash 0342b54bcbd4aa1679bf23b2a4cc8177f630e6415b7a82e58cefb9f8991e0a8e

flyn-org commented:
@neheb, I updated PKG_MIRROR_HASH.
